DataFormats/Provenance/test/indexIntoFile2_t.cppunit.cc

```diff
@@ -445,3 +474,374 @@ void TestIndexIntoFile2::testAddEntryAndFixAndSort() {
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[11].lumi() == 1);
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[11].beginEvents() == 5);
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[11].endEvents() == 6);
+
+  indexIntoFile.addEntry(fakePHID2, 2, 2, 2, 7);  // Event
+  indexIntoFile.addEntry(fakePHID2, 2, 2, 0, 5);  // Lumi
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries().size() == 13);
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[12].processHistoryIDIndex() == 1);
+  CPPUNIT_ASSERT(indexIntoFile.processHistoryIDs().size() == 2);
+  CPPUNIT_ASSERT(indexIntoFile.processHistoryIDs()[1] == fakePHID2);
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[12].orderPHIDRun() == IndexIntoFile::invalidEntry);
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[12].orderPHIDRunLumi() == 6);
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[12].entry() == 5);
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[12].run() == 2);
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[12].lumi() == 2);
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[12].beginEvents() == 6);
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[12].endEvents() == 8);
+
+  indexIntoFile.addEntry(fakePHID1, 1, 1, 0, 6);  // Lumi
+  indexIntoFile.addEntry(fakePHID2, 1, 1, 0, 7);  // Lumi
+  indexIntoFile.addEntry(fakePHID2, 2, 1, 0, 8);  // Lumi
+
+  indexIntoFile.addEntry(fakePHID1, 11, 0, 0, 1);  // Run
+  indexIntoFile.addEntry(fakePHID1, 1, 0, 0, 2);   // Run
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[17].orderPHIDRun() == 1);
+  indexIntoFile.addEntry(fakePHID2, 1, 0, 0, 3);  // Run
+
+  indexIntoFile.addEntry(fakePHID2, 2, 0, 0, 4);  // Run
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries().size() == 20);
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[19].processHistoryIDIndex() == 1);
+  CPPUNIT_ASSERT(indexIntoFile.processHistoryIDs().size() == 2);
+  CPPUNIT_ASSERT(indexIntoFile.processHistoryIDs()[1] == fakePHID2);
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[19].orderPHIDRun() == 5);
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[19].orderPHIDRunLumi() == IndexIntoFile::invalidEntry);
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[19].entry() == 4);
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[19].run() == 2);
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[19].lumi() == IndexIntoFile::invalidLumi);
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[19].beginEvents() == IndexIntoFile::invalidEntry);
+  CPPUNIT_ASSERT(indexIntoFile.runOrLumiEntries()[19].endEvents() == IndexIntoFile::invalidEntry);
+
+  indexIntoFile.addEntry(fakePHID1, 2, 0, 0, 5);  // Run
+
+  indexIntoFile.sortVector_Run_Or_Lumi_Entries();
+
+  int index = 0;
+  for (auto const& entry : indexIntoFile.runOrLumiEntries()) {
+    if (index == 0) {
+      CPPUNIT_ASSERT(entry.orderPHIDRun() == 0);
+      CPPUNIT_ASSERT(entry.orderPHIDRunLumi() == IndexIntoFile::invalidEntry);
+      CPPUNIT_ASSERT(entry.entry() == 1);
+    }
+    if (index == 1) {
```

Comment:
How about `else if`, or `switch`?

Or organizing the code along
```cpp
auto entry = indexIntoFile.runOrLumiEntries().begin();

CPPUNIT_ASSERT(entry.orderPHIDRun() == 0);
CPPUNIT_ASSERT(entry.orderPHIDRunLumi() == IndexIntoFile::invalidEntry);
CPPUNIT_ASSERT(entry.entry() == 1);

++entry;
CPPUNIT_ASSERT(entry.orderPHIDRun() == 0);
CPPUNIT_ASSERT(entry.orderPHIDRunLumi() == 0);
CPPUNIT_ASSERT(entry.entry() == 0);

++entry
...
CPPUNIT_ASSERT(entry == indexIntoFile.runOrLumiEntries().end());
```
?

Ok, I see the "for-switch" as a general pattern in these unit tests that existed already. So maybe it is better to be consistent. I'm not sure if changing the pattern is really worth of the effort (in a separate PR).

Reply:
Yes. I agree. Switch would have been a better choice. I probably established this anti-pattern long ago (and then copied it multiple times). If you think it is worth the effort I will change it now or later in another PR...

Reply:
I thought about this a little more over the weekend. This is easy to fix and will not take long. I'll do it in a separate PR after this PR is merged. Do you have a preference of the 3 choices you suggested? In some of the cases the test occurs on only some of the entries, so switch or else if is better. For many of them any of the three choices would be fine... I don't really have a preference. I'll go with switch if you don't have a preference.